src/js/components/modal.js

```diff
@@ -0,0 +1,265 @@
+const CLOSERS = `${CLOSE_BUTTON}, .${OVERLAY_CLASSNAME}:not([${FORCE_ACTION_ATTRIBUTE}])`;
+
+const ACTIVE_CLASS = "usa-js-mobile-nav--active";
+const PREVENT_CLICK_CLASS = "usa-js-no-click";
+const VISIBLE_CLASS = "is-visible";
+const HIDDEN_CLASS = "is-hidden";
+
+const nonModals = document.querySelectorAll(
+  `body > *:not(${MODAL}):not([aria-hidden])`
+);
+
+let modal;
+
+const isActive = () => document.body.classList.contains(ACTIVE_CLASS);
+const SCROLLBAR_WIDTH = ScrollBarWidth();
+const INITIAL_PADDING = window
+  .getComputedStyle(document.body)
+  .getPropertyValue("padding-right");
+const TEMPORARY_PADDING =
+  parseInt(INITIAL_PADDING.replace(/px/, ""), 10) +
+  parseInt(SCROLLBAR_WIDTH.replace(/px/, ""), 10) +
+  "px";
+
+/**
+ *  Is bound to escape key, closes modal when
+ */
+const onMenuClose = () => {
+  modal.toggleModal.call(modal, false);
+};
+
+/**
+ *  Toggle the visibility of a modal window
+ *
+ * @param {KeyboardEvent} event the keydown event
+ * @returns {boolean} safeActive if mobile is open
+ */
+function toggleModal(event) {
+  let originalOpener;
+  let clickedElement = event.target;
+  const { body } = document;
+  const safeActive = !isActive();
+  const modalId = clickedElement
+    ? clickedElement.getAttribute("aria-controls")
+    : document.querySelector(".usa-modal-wrapper.is-visible");
+  const targetModal = safeActive
+    ? document.getElementById(modalId)
+    : document.querySelector(".usa-modal-wrapper.is-visible");
+  const openFocusEl = targetModal.querySelector(INITIAL_FOCUS)
+    ? targetModal.querySelector(INITIAL_FOCUS)
+    : targetModal.querySelector(".usa-modal");
+  const returnFocus = document.getElementById(
+    targetModal.getAttribute("data-opener")
+  );
+  const menuButton = body.querySelector(OPENERS);
+  const forceUserAction = targetModal.getAttribute(FORCE_ACTION_ATTRIBUTE);
+
+  // Sets the clicked element to the close button
+  // so esc key always closes modal
+  if (event.type === "keydown" && targetModal !== null) {
+    clickedElement = targetModal.querySelector(CLOSE_BUTTON);
+  }
+
+  // When we're not hitting the escape key…
+  if (clickedElement) {
+    // Make sure we click the opener
+    // If it doesn't have an ID, make one
+    // Store id as data attribute on modal
+    if (clickedElement.hasAttribute(OPENER_ATTRIBUTE)) {
+      if (this.getAttribute("id") === null) {
+        originalOpener = `modal-${Math.floor(Math.random() * 900000) + 100000}`;
+        this.setAttribute("id", originalOpener);
+      } else {
+        originalOpener = this.getAttribute("id");
+      }
+      targetModal.setAttribute("data-opener", originalOpener);
+    }
+
+    // This basically stops the propagation if the element
+    // is inside the modal and not a close button or
+    // element inside a close button
+    if (clickedElement.closest(`.${MODAL_CLASSNAME}`)) {
+      if (clickedElement.hasAttribute(CLOSER_ATTRIBUTE) || 
```

Comment:
Had to revert back to writing the if statement like this. VoiceOver didn't seem to like: 

```js
    if (
      clickedElement.closest(`.${MODAL_CLASSNAME}`) &&
      (!clickedElement.hasAttribute(CLOSER_ATTRIBUTE) ||
        !clickedElement.closest(`[${CLOSER_ATTRIBUTE}]`))
    ) {
      event.stopPropagation();
      return false;
    }
  ```